src/handle_get_printable_amount.c

```diff
@@ -7,32 +7,32 @@
 #include "common_utils.h"
 #include "uint256.h"
 #include "string.h"
+#include "network.h"
 
 void handle_get_printable_amount(get_printable_amount_parameters_t* params,
                                  chain_config_t* config) {
-    uint8_t decimals;
     char ticker[MAX_TICKER_LEN];
+    uint8_t decimals;
+    uint64_t chain_id = 0;
+
     memset(params->printable_amount, 0, sizeof(params->printable_amount));
     if (params->amount_length > 32) {
         PRINTF("Amount is too big, 32 bytes max but buffer has %u bytes", params->amount_length);
         return;
     }
 
+    if (!parse_swap_config(params->coin_configuration,
+                           params->coin_configuration_length,
+                           ticker,
+                           &decimals,
+                           &chain_id)) {
+        PRINTF("Error while parsing config\n");
+        return;
+    }
     // If the amount is a fee, its value is nominated in ETH even if we're doing an ERC20 swap
     if (params->is_fee) {
-        uint8_t ticker_len = strnlen(config->coinName, sizeof(config->coinName));
-        memcpy(ticker, config->coinName, ticker_len);
-        ticker[ticker_len] = '\0';
+        strlcpy(ticker, get_displayable_ticker(&chain_id, config), sizeof(ticker));
```

Comment:
can get_displayable_ticker fail ?

Reply:
It cannot, there is always the backup from the given config.

Reply:
A better solution will be looked at in a future PR :+1: